### PR TITLE
Adds a warning to whisper succumb

### PR DESCRIPTION
**Pull request by ynot01**
### Intent of your Pull Request

`You will be immediately killed by this action. Proceed?`

#### Changelog

:cl:  
rscadd: Added a warning to whisper succumb
/:cl:

### CODE REVIEW

File: code/modules/mob/living/say.dm [hunk 1]
```diff
@@ -158,6 +158,10 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 
 	var/fullcrit = InFullCritical()
 	if((InCritical() && !fullcrit) || message_mode == MODE_WHISPER)
+		if(fullcrit)
+			var/alertresult = alert(src, "You will be immediately killed by this action. Proceed?",,"Accept", "Decline")
+			if(alertresult == "Decline" || QDELETED(src) || !istype(src) || !src.key)
```
Comment (alexkar598) [inline code/modules/mob/living/say.dm]:
```suggestion
			if(alertresult == "Decline" || QDELETED(src))
```
1) You cant not be yourself
2) You cant be disconnected and answer an alert

Comment (alexkar598) [inline code/modules/mob/living/say.dm]: Run the critical checks again, otherwise we will have people succumbing by saving the prompt for later lol

**Review by alexkar598 — DISMISSED** [review]

File: code/modules/mob/living/say.dm (hunk 1 above)
Comment (ynot01) [reply code/modules/mob/living/say.dm]: wait you want me to run an && if fullcrit again directly after the prompt

Comment (ynot01) [reply code/modules/mob/living/say.dm]: wait wait i think i get what you mean, it already does run a if fullcrit a few lines down
here's the entire thing
```
	if((InCritical() && !fullcrit) || message_mode == MODE_WHISPER)
		if(fullcrit)
			var/alertresult = alert(src, "You will be immediately killed by this action. Proceed?",,"Accept", "Decline")
			if(alertresult == "Decline" || QDELETED(src))
				return FALSE
		message_range = 1
		message_mode = MODE_WHISPER
		src.log_talk(message, LOG_WHISPER)
		if(fullcrit)
			var/health_diff = round(-HEALTH_THRESHOLD_DEAD + health)
			// If we cut our message short, abruptly end it with a-..
			var/message_len = length_char(message)
			message = copytext_char(message, 1, health_diff) + "[message_len > health_diff ? "-.." : "..."]"
			message = Ellipsis(message, 10, 1)
			last_words = message
			message_mode = MODE_WHISPER_CRIT
			succumbed = TRUE
	else
		src.log_talk(message, LOG_SAY, forced_by=forced)
```

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**